src/app_feup/lib/controller/library/library.dart

```diff
@@ -27,40 +28,75 @@ class Library implements LibraryInterface {
   Cookie pdsCookie;
   String _username;
   String _password;
-
-  List<String> faculties;
+  String faculty;
 
   /**
    * Need to create a factory function since we need the constructor
    *  to have an async method
    */
-  static Future<Library> create() async {
+  static Future<Library> create({Cookie pdsCookie}) async {
     final library = Library();
+    if (pdsCookie != null) {
+      library.pdsCookie = pdsCookie; // Set libraries' pds cookie if exists
+    }
 
-    library.faculties = await AppSharedPreferences.getUserFaculties();
+    try {
+      library.faculty = (await AppSharedPreferences.getUserFaculties()).first;
+    } catch (err) {
+      Logger().e("Unable to retrieve the user's institution");
+    }
 
     return library;
   }
 
+  /**
+   * Wrapper to send a request, after first getting the aleph cookie
+   * pdsCookie is an optional argument
+   */
+  static Future<http.Response> libRequestWithAleph(String url,
+      {Cookie pdsCookie = null}) async {
```

Comment:
```suggestion
      {Cookie pdsCookie}) async {
```
I think this does the same